app/services/sso/openid_connect.rb

```diff
@@ -38,7 +38,16 @@ def bearer_token_set?
 
     def valid_issuer?
       payload = jwt_token.decoded_payload
-      payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+      unless payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+        logger = Foreman::Logging.logger('app')
```

Comment:
As a side note, a good pattern is

```ruby
delegate :logger, to: :Rails
```

so you can simply type `logger.debug` in the class as in a model.